src/App.js

```diff
@@ -1,7 +1,37 @@
-import "./App.css";
+import React from "react";
+import ItemList from "./Component/ItemList";
+import CheckOut from "./Component/CheckOut";
+import Cart from "./Component/Cart"
 
-const App = () => {
-  return <h1>Hello, world!</h1>;
-};
 
-export default App;
+
+class App extends React.Component {
+   state ={cart:[], subtotal:0}
+
+
+ addCart = (item) => {
+   this.setState((prevState) => {
+    this.subtotal(item.price)
+     return {cart:[item, ...prevState.cart]}
+   })
+ }
+ subtotal = (tot) => {
```

Comment:
Because subtotal can be calculated based on the items in this.state.cart, you should move the logic to calculate the subtotal to the beginning of your render method.